File: src/main/java/net/kenro/ji/jin/purescript/psi/impl/PSNewTypeDeclarationImpl.java
package net.kenro.ji.jin.purescript.psi.impl;

import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import net.kenro.ji.jin.purescript.psi.PSNewTypeDeclaration;
import net.kenro.ji.jin.purescript.psi.PSVisitor;
import org.jetbrains.annotations.NotNull;

public class PSNewTypeDeclarationImpl extends PSPsiElement implements PSNewTypeDeclaration {

    public PSNewTypeDeclarationImpl(final ASTNode node) {
        super(node);
    }

    public void accept(@NotNull final PSVisitor visitor) {
        visitor.visitPSNewTypeDeclaration(this);
    }

    public void accept(@NotNull final PsiElementVisitor visitor) {
        if (visitor instanceof PSVisitor) accept((PSVisitor) visitor);
        else super.accept(visitor);
    }





}

File: src/main/java/net/kenro/ji/jin/purescript/psi/impl/PSNewTypeDeclarationImpl.kt
package net.kenro.ji.jin.purescript.psi.impl

import com.intellij.lang.ASTNode
import net.kenro.ji.jin.purescript.psi.PSNewTypeDeclaration

class PSNewTypeDeclarationImpl(node: ASTNode) : PSPsiElement(node),
    PSNewTypeDeclaration